src/routes/callback/callback.ts

```diff
@@ -4,3 +4,13 @@ export function retrieveCode() {
 	const code = urlParams.get('code');
 	return code;
 }
+
+export async function storeUserData(name: string, id: string) {
+	const storedValueString: string | null = localStorage.getItem('ViBE');
+	if (storedValueString) {
+		const storedValue = JSON.parse(storedValueString);
+		storedValue.display_name = name;
+		storedValue.user_id = id;
+		localStorage.setItem('ViBE', JSON.stringify(storedValue));
+	}
+}
```

Comment:
This is a critical part of the app, which means more tests for every possible edge case.

Additionally, need to set up try catch statements. You are learning a lot of technical skills, but lacking in prioritizing user experience - like what happens when user is stranded on a broken site?

1. inform customer when shit hits the fan
2. log it so when shit does hit the fan, yk its real.

Reply:
will update